### PR TITLE
Copy all the data-driven form files into Calypso

### PR DESCRIPTION
**Pull request by DanReyLop**
Closes https://github.com/automattic/wp-calypso/issues/19521

This copies all the data-driven form files that were originally in the `client/apps/settings` folder on the `woocommerce-services` repo into Calypso.

The import paths have been fixed, and the required generic files (such as components) have been copied as well, but there was no more work beyond that. This PR should be a baseline for the rest of the feature, I didn't want to make it even bigger.

In the process of bringing everything to Calypso, I removed the parts that don't apply to the Shipping Services form, such as toggles, indicators or textareas (they're only used in the WP-Admin self-help page.

### CODE REVIEW

File: client/extensions/woocommerce/woocommerce-services/state/service-settings/values/actions.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+export const submit = ( schema, silent ) => ( dispatch, getState, { methodId, instanceId, formType } ) => {
+	silent = ( true === silent );
+
+	const setIsSaving = ( value ) => dispatch( FormActions.setFormProperty( 'isSaving', value ) );
+
+	const setSuccess = ( value ) => {
+		dispatch( FormActions.setFormProperty( 'success', value ) );
+		if ( ! silent && true === value ) {
+			dispatch( NoticeActions.successNotice( translate( 'Your changes have been saved.' ), {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/state/service-settings/values/actions.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Your changes have been saved.', { context: 'voce settings api'} )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/copy/wcs-data-friven-form)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by allendav** [comment]
Just curious - why are we making this change?

**Comment by DanReyLop** [comment]
@allendav Same as with the other branches that port functionality from WCS to Calypso, I decided to first copy all the code and then make it work, we figured it was easier to review. So this is the base branch, and there are *a bunch* of PRs depending on this (I've just flooded the #store channel with the last ones).

**Comment by DanReyLop** [comment]
Every branch for this feature has been collapsed into this PR. This is the branch to test and do a final design/code pass.

@allendav @marcinbot Thanks for your reviews! I don't think it's worth it to review all the code changes again, what do you folks think?

@kellychoffman This is the current design:
usps-list
usps-modal

It behaves the same way as WP-Admin: When the modal is opened no errors are immediately present to the user. Only when the user interacts with a field, the field is validated and an error shown. When the user clicks the "Save" button, a full validation is performed so he's not allowed to exit the dialog unless everything is correct (or he cancels the operation).

The rest of the flow works exactly like the other shipping methods in Calypso: eerything needed to interact with the UI is preloaded, all the interactions happen locally, and when the user clicks "Save changes" everything is committed to the server.

**Comment by kellychoffman** [comment]
Just one comment on the design in the second screenshot: 

The spacing that you see below "Services" and above "Which rates would you like to show?" sections should be applied to all other form fields.

**Comment by DanReyLop** [comment]
@kellychoffman Fixed in https://github.com/Automattic/wp-calypso/pull/20908/commits/a2e76253351dd4aef052c66f54d0e9485ac41f1c

**Comment by DanReyLop** [comment]
I've made some fixes for errors spotted by Flow Patrol, and I updated this branch with `master`. This is ready for a final pass :)

File: client/extensions/woocommerce/state/sites/shipping-methods/selectors.js [hunk 2]
```diff
@@ -3,15 +3,30 @@
...
+	isShippingMethodSchemaLoaded,
+	isShippingMethodSchemaLoading,
+} from 'woocommerce/woocommerce-services/state/shipping-method-schemas/selectors';
+
+/*
+ * By default, those methods are called "XXXX (WooCommerce Services)".
+ * We aren't pushing the "WooCommerce Services" brand anywhere in Calypso, so the method names must be changed.
+ */
+const METHOD_NAMES = {
+	wc_services_usps: translate( 'USPS', { comment: 'United States Postal Services' } ),
```
Comment (akirk) [inline client/extensions/woocommerce/state/sites/shipping-methods/selectors.js]: Is there really a point in translating these? They are names of companies.

Comment (DanReyLop) [reply client/extensions/woocommerce/state/sites/shipping-methods/selectors.js]: Not really. I'll remove it in a later PR, you got here 1 minute too late :)

Comment (akirk) [reply client/extensions/woocommerce/state/sites/shipping-methods/selectors.js]: Close call. Thanks!

**Pull request merged**